pom.xml

```diff
@@ -42,6 +42,11 @@
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>jackson2-api</artifactId>
+            <scope>test</scope>
+        </dependency>
```

Comment:
Why is this here?

Reply:
if not we got this 
```
[WARNING] Rule 5: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for com.fasterxml.jackson.datatype:jackson-datatype-json-org:2.9.8 paths to dependency are:
+-io.jenkins.plugins:file-parameters:1.0-SNAPSHOT
  +-org.jenkinsci.plugins:pipeline-model-definition:1.7.2
    +-org.jenkinsci.plugins:pipeline-model-api:1.7.2
      +-com.fasterxml.jackson.datatype:jackson-datatype-json-org:2.9.8
and
+-io.jenkins.plugins:file-parameters:1.0-SNAPSHOT
  +-org.jenkinsci.plugins:pipeline-model-definition:1.7.2
    +-org.jenkinsci.plugins:pipeline-model-api:1.7.2
      +-org.jenkins-ci.plugins:jackson2-api:2.12.1 (managed) <-- org.jenkins-ci.plugins:jackson2-api:2.9.8
        +-com.fasterxml.jackson.datatype:jackson-datatype-json-org:2.12.1
, 
Require upper bound dependencies error for org.json:json:20171018 paths to dependency are:
+-io.jenkins.plugins:file-parameters:1.0-SNAPSHOT
  +-org.jenkinsci.plugins:pipeline-model-definition:1.7.2
    +-org.jenkinsci.plugins:pipeline-model-api:1.7.2
      +-com.fasterxml.jackson.datatype:jackson-datatype-json-org:2.9.8
        +-org.json:json:20171018
and
+-io.jenkins.plugins:file-parameters:1.0-SNAPSHOT
  +-org.jenkinsci.plugins:pipeline-model-definition:1.7.2
    +-org.jenkins-ci.plugins:git-client:3.6.0
      +-org.jenkins-ci.plugins:trilead-api:1.0.13
        +-org.jenkins-ci:trilead-ssh2:build-217-jenkins-27
          +-com.google.crypto.tink:tink:1.5.0
            +-org.json:json:20180813
```

Reply:
Hmm @timja sounds like a `bom` failure. Sound familiar? `sample-plugin/pom.xml` includes this dep, which it ought not.

Reply:
are you saying that `sample-plugin` shouldn't have jackson2-api @jglick ?

Reply:
Well it should not _need_ it, and it looks like the fact that it does have that explicit dependency is masking a problem.

Reply:
I guess it will be fixed when this change https://github.com/jenkinsci/pipeline-model-definition-plugin/commit/2c186f56aa74557c77b5f1be0d6dcd045680eaef will be release and then new version integrated in the bom
but until all of this done we need this dependency (as `test`) or this:
```
        <dependency>
            <groupId>org.jenkinsci.plugins</groupId>
            <artifactId>pipeline-model-definition</artifactId>
            <scope>test</scope>
            <exclusions>
                <exclusion>
                    <groupId>com.fasterxml.jackson.datatype</groupId>
                    <artifactId>jackson-datatype-json-org</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
```
personally I do not have any strong opinion on what workaround to use as they are bot `test` dependencies.